### PR TITLE
Multiple attribute-surrounding expressions

### PR DESCRIPTION
**Pull request by duncanbeevers**
customAttrOpen / customAttrClose replaced by customAttrSurround

customAttrSurround is an array of tuples.
Each tuple represents an open/close regexp pair.

### CODE REVIEW

**Comment by duncanbeevers** [comment]
This should make the CLI integration simpler as the functionality now accepts arrays of attribute-surrounding open/close regexp pairs.

Related to #189 #201

@toxinhead I tested this with your `/\{\%[^\}]+%\}/`, `/\{\%[^\}]+%\}/` regexp pair and it seemed to work fine. If you still have an issue, open an issue either here, or in kangax/html-minifier if this gets merged, and I'll add an explicit test-case for it.

**Comment by boycce** [comment]
Hey @duncanbeevers, I created a fix & test for my problem when passing htmlMinifyOptions with 
`<div onc=""></div>`

https://github.com/toxinhead/html-minifier/commit/46556e13115827c988cc878012a827e29c5a5b9f

File: dist/htmlminifier.js [hunk 1]
```diff
@@ -68,41 +68,62 @@
...
-
-    var startTag, attr;
+  function startTagForHandler( handler ) {
     var customStartTagAttrs;
-    if (handler.customAttrOpen && handler.customAttrClose) {
+
+    if ( handler.customAttrSurround ) {
+      var attrClauses = [];
+
+      for ( var i = handler.customAttrSurround.length - 1; i >= 0; i-- ) {
```
Comment (kangax) [inline dist/htmlminifier.js:77]: Btw, this could just be `for ( var i = handler.customAttrSurround.length - 1; i--; ) {`

**Comment by kangax** [comment]
Thanks @duncanbeevers!

File: dist/htmlminifier.js (hunk 1 above)
Comment (duncanbeevers) [inline dist/htmlminifier.js:77]: Ooh, I like that.

**Pull request merged**